src/components/NodeSelector.tsx

```diff
@@ -0,0 +1,396 @@
+import { Icon } from '@iconify/react';
+import {
+  Autocomplete,
+  Box,
+  Button,
+  Checkbox,
+  Chip,
+  Collapse,
+  FormControl,
+  FormControlLabel,
+  FormHelperText,
+  Stack,
+  Switch,
+  TextField,
+  Typography,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { fetchAvailableNodes } from './chatUtils';
+
+interface NodeInfo {
+  name: string;
+  labels: Record<string, string>;
+  ready: boolean;
+  taints: Array<{
+    key: string;
+    value?: string;
+    effect: string;
+  }>;
+}
+
+interface NodeSelectorProps {
+  selectedNodes: string[];
+  onNodesChange: (_nodes: string[]) => void;
+  labelSelector?: string;
+  onLabelSelectorChange?: (_selector: string) => void;
+  disabled?: boolean;
+  showLabelSelector?: boolean;
+  helperText?: string;
+  onRequiredNodesChange?: (
+    _requiredNodes: number | '',
+    _isExactMatch: boolean,
+    _willAutoProvision: boolean
+  ) => void;
+}
+
+const COMMON_LABEL_SELECTORS = [
+  { label: 'GPU Nodes', value: 'accelerator=nvidia' },
+  { label: 'AMD64 Architecture', value: 'kubernetes.io/arch=amd64' },
```

Comment:
I don't think we need this. The model should still be able to deploy regardless of the CPU architecture